base/abstractset.jl

```diff
@@ -117,9 +117,28 @@ Set{Int64} with 1 element:
   2
 ```
 """
-intersect(s::AbstractSet, itr, itrs...) = intersect!(intersect(s, itr), itrs...)
+function intersect(s::AbstractSet, itr, itrs...)
+    # determine if swap order is useful and viable
+    if length(s)>50 && haslength(itr) && all(map(haslength, itrs))
+        itrs_lengths = map(length, itrs)
+        min_idx = argmin(itrs_lengths)
+        min_length = itrs_lengths[min_idx]
+        # do nothing if s is longer than the rest or itr is already the shortest
+        if length(itr) > min_length >= length(s)
```

Comment:
Why is this looking at `length(s)`? Why not always intersect `s` with the shortest itr?